test/metabase/upload_test.clj

```diff
@@ -189,13 +217,39 @@
             (is (=? {:name      #"(?i)number"
                      :base_type :type/Float}
                     (t2/select-one Field :database_position 4 :table_id (:id table))))
+            (is (=? {:name      #"(?i)date"
+                     :base_type :type/Date}
+                    (t2/select-one Field :database_position 5 :table_id (:id table))))
+            (is (=? {:name      #"(?i)datetime"
+                     :base_type :type/DateTime}
+                    (t2/select-one Field :database_position 6 :table_id (:id table))))
             (testing "Check the data was uploaded into the table"
               (is (= [[2]] (-> (mt/process-query {:database (mt/id)
                                                   :type :query
                                                   :query {:source-table (:id table)
                                                           :aggregation [[:count]]}})
                                mt/rows))))))))))
 
+(deftest load-from-csv-date-test
```

Comment:
This is pretty redundant because of the previous test, but I thought I'd create a new `deftest` to exhaustively test different date formats when we get to that.